### ISSUE TITLE
CloudKit Seamless Authentication Between User Devices

### ISSUE THREAD

**Issue by TimOliver** [realm/RealmTasks#10]
One of the overarching goals of this app is to consider each aspect of the experience from the user's perspective. Obviously, one of the more challenging aspects of Sync is coming up with a pattern where Sync is able to determine all of the devices a user owns (And would like their content sync'd to), with as little interaction on the user's behalf, and certainly avoiding requiring private information (Like email addresses).

In the actual Clear app, activating sync via iCloud is a simple as a single tap in the Settings dialog:
![clearsync](https://cloud.githubusercontent.com/assets/429119/16276968/a43a1bf8-3865-11e6-86aa-606f1c8a627e.jpg)

This is the bar we should aim for, as anything lower than this would obviously be an inferior experience for the user.

An interesting article discovered by @astigsen shows a really good way of achieving this: https://medium.com/@skreutzb/ios-onboarding-without-signup-screens-cb7a76d01d6e#.rpxuwq3ke

The idea is to use CloudKit as a seamless authentication layer in order to identify users via their iCloud account, but still maintaining a complete level of anonymity without directly requiring their account credentials.

In a nutshell, CloudKit provides each app with a token that is unique for that user and their Apple ID, guaranteed to be the same for the app running on each of their devices.

In this way, it would simply be a matter of requesting this iCloud token on initial launch of the app, and then using it as the user identity value for Realm Sync.

From the user perspective, this would appear completely seamless and automatic.
